### PR TITLE
doc: Correct a typo in project.md

### PR DESCRIPTION
**Pull request by doolio**
## Pull Request Checklist

~~- [ ] A news fragment is added in `news/` describing what is new.~~
~~- [ ] Test cases added for changed code.~~

## Describe what you have changed in this PR.
Corrected a typo in project.md.
Added change to commit #2210 requested by @pawamoy after review.

### CODE REVIEW

File: docs/docs/usage/project.md [hunk 1]
```diff
@@ -74,7 +74,7 @@ Although PDM run on Python 3.7 and above, you can still have lower Python versio
 | `flit-core>=3.4`      | `>=3.6`          | Yes             |
 | `flit-core>=3.2,<3.4` | `>=3.4`          | Yes             |
 
-Note that if your project is an application(without `name` metadata), the above limitation of backends don't apply, since you don't need a build backend after all, and you can use a Python version up to `2.7`.
+Note that if your project is an application(without `name` metadata), the above limitation of backends don't apply, since you don't need a build backend after all, and you can use a Python version up to `3.7`.
```
[the remark this answers is not in the document]
Comment (pawamoy) [reply docs/docs/usage/project.md:77]: I guess so :thinking:

[the remark this answers is not in the document]
Comment (doolio) [reply docs/docs/usage/project.md:77]: Then "down to 2.7" is more accurate. I'll reword that paragraph to improve the general wording and take that constraint into account.

[the remark this answers is not in the document]
Comment (doolio) [reply docs/docs/usage/project.md]: Done. Let me know if you prefer I rebase onto `main`.

File: docs/docs/usage/project.md [hunk 2]
```diff
@@ -74,7 +74,8 @@ Although PDM run on Python 3.7 and above, you can still have lower Python versio
 | `flit-core>=3.4`      | `>=3.6`          | Yes             |
 | `flit-core>=3.2,<3.4` | `>=3.4`          | Yes             |
 
-Note that if your project is an application(without `name` metadata), the above limitation of backends don't apply, since you don't need a build backend after all, and you can use a Python version up to `2.7`.
+Note that if your project is an application (i.e. without the `name` metadata),
+the above limitation of backends does not apply. Therefore, if you don't need a build backend you can use any Python version `≥2.7`.
```
Comment (pawamoy) [inline docs/docs/usage/project.md:78]: For consistency:

```suggestion
the above limitation of backends does not apply. Therefore, if you don't need a build backend you can use any Python version `>=2.7`.
```

File: docs/docs/usage/project.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (doolio) [reply docs/docs/usage/project.md]: The only thing left to consider with this section is that it contradicts the fact that one needs >= 3.7 to use PDM in the first place.

[the remark this answers is not in the document]
Comment (pawamoy) [reply docs/docs/usage/project.md]: PDM runs on Python >= 3.7, but an application project can support Python >= 2.7, and you'll still be able to manage its dependencies with PDM.

**Review by pawamoy — APPROVED** [review]

**Pull request merged**